src/pathfinder/water_regions.cpp

```diff
@@ -196,7 +226,7 @@ class WaterRegion
 		for (int y = 0; y < WATER_REGION_EDGE_LENGTH; ++y) {
 			std::string line{};
 			for (int x = 0; x < WATER_REGION_EDGE_LENGTH; ++x) {
-				const auto label = this->tile_patch_labels[x + y * WATER_REGION_EDGE_LENGTH];
+				const auto label = this->GetLabel(TileAddXY(tile_area.tile, x, y));
```

Comment:
TileAddXY would increment the index with the map's X-size for each increment in the Y direction, resulting in out of bounds.

```suggestion
				const auto label = this->GetLabel(GetLocalIndex(TileAddXY(tile_area.tile, x, y)));
```

Reply:
GetLabel already calls GetLocalIndex if required.

Reply:
Oh you're right. Ignore my comment then :)